linkerd/app/inbound/src/lib.rs

```diff
@@ -290,11 +292,16 @@ impl<A: OrigDstAddr> Config<A> {
                 http_server.into_inner(),
                 h2_settings,
                 drain.clone(),
-                disable_protocol_detection_for_ports.clone(),
             );
 
+            let svc = svc::stack(tcp_server)
+                .push(DetectProtocolLayer::new(ProtocolDetect::new(
+                    disable_protocol_detection_for_ports.clone(),
+                )))
+                .push_timeout(detect_protocol_timeout);
+
             // Terminate inbound mTLS from other outbound proxies.
-            let accept = tls::AcceptTls::new(local_identity, tcp_server)
+            let accept = tls::AcceptTls::new(local_identity, svc.into_inner())
```

Comment:
Don't we need to timeout outside of the TLS detection? Or, why don't we encounter this problem with TLS detection?